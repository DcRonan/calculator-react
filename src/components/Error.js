const Error = () => <h4>Error! Page not found.</h4>;

export default Error;
