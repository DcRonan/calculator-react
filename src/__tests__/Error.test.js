import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Error from '../components/Error';

describe('checks if Error component renders correctly', () => {
  test('should renders correct text', () => {
    render(<Error />);
    const text = screen.getByText('Error! Page not found.');
    expect(text).toBeInTheDocument();
  });

  test('should render correctly', () => {
    const tree = renderer.create(<Error />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
