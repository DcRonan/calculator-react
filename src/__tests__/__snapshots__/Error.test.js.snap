// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks if Error component renders correctly should render correctly 1`] = `
<h4>
  Error! Page not found.
</h4>
`;
